op-acceptance-tests/tests/flashblocks/flashblocks_stream_test.go

```diff
@@ -51,38 +59,103 @@ func TestFlashblocksStream(gt *testing.T) {
+	testDuration := time.Duration(int64(flashblocksStreamRateMs*maxExpectedFlashblocks*2)) * time.Millisecond
+	failureTolerance := int(0.15 * float64(maxExpectedFlashblocks))
+
+	logger.Debug("Test duration", "duration", testDuration, "failure tolerance (of flashblocks)", failureTolerance)
+
+	require.NotNil(t, rollupBoostNode.Escape(), "rollupBoostNode should not be nil")
+
+	// Instrument builder stream separately to confirm flashblocks emission upstream.
+	builderOutput := make(chan []byte, maxExpectedFlashblocks)
+	builderDone := make(chan struct{})
+	go func() {
+		err := oprbuilderNode.FlashblocksClient().ListenFor(logger.With("stream_source", "op-rbuilder"), testDuration, builderOutput, builderDone)
+		require.NoError(t, err)
+	}()
+	builderMessages := make([]string, 0)
+
+	output := make(chan []byte, maxExpectedFlashblocks)
+	doneListening := make(chan struct{})
+	streamedMessages := make([]string, 0)
+	go func() {
+		err := rollupBoostNode.FlashblocksClient().ListenFor(logger.With("stream_source", "rollup-boost"), testDuration, output, doneListening)
+		require.NoError(t, err)
+	}()
+
+	listening := true
+	for listening {
+		select {
+		case <-doneListening:
+			doneListening = nil
+		case <-builderDone:
+			builderDone = nil
+		case msg := <-output:
+			streamedMessages = append(streamedMessages, string(msg))
+		case msg := <-builderOutput:
+			builderMessages = append(builderMessages, string(msg))
+		}
 
-				mode := FlashblocksStreamMode_Follower
-				if dsl.NewConductor(flashblocksBuilderNode.Escape().Conductor()).IsLeader() {
-					mode = FlashblocksStreamMode_Leader
-				}
+		if doneListening == nil && builderDone == nil {
+			listening = false
+		}
+	}
 
-				testFlashblocksStreamRbuilder(tt, logger, flashblocksBuilderNode, mode, flashblocksStreamRateMs)
-			}
+	defer close(output)
+	defer close(builderOutput)
 
-			for _, flashblocksWebsocketProxy := range sys.FlashblocksWebsocketProxies[l2Chain] {
-				testFlashblocksStreamFbWsProxy(tt, logger, flashblocksWebsocketProxy, flashblocksStreamRateMs)
-			}
-		})
+	logger.Info("Completed WebSocket stream reading", "message_count", len(streamedMessages))
+	logger.Info("Completed builder WebSocket stream reading", "message_count", len(builderMessages))
+
+	if len(builderMessages) > 0 {
+		logger.Info("Sample builder message", "payload", builderMessages[0])
 	}
+
+	totalFlashblocksProduced := evaluateFlashblocksStream(t, logger, streamedMessages, failureTolerance)
+	require.Greater(t, totalFlashblocksProduced, 0, "expected to receive flashblocks from rollup-boost stream")
+	logger.Info("Flashblocks stream validation completed", "total_flashblocks_produced", totalFlashblocksProduced)
+}
+
+// driveViaTestSequencer explicitly builds a few blocks to ensure the builder/rollup-boost
+// have payloads to serve before we start listening for flashblocks.
+func driveViaTestSequencer(t devtest.T, sys *presets.SingleChainWithFlashblocks, count int) {
```

Comment:
How is this different than just waiting for the L2EL to produce some blocks? Is it faster?